cmd/promtool/unittest.go

```diff
@@ -158,6 +172,33 @@ func resolveAndGlobFilepaths(baseDir string, utf *unitTestFile) error {
 			fmt.Fprintln(os.Stderr, "  WARNING: no file match pattern", rf)
 		}
 		globbedFiles = append(globbedFiles, m...)
+		for i, file := range globbedFiles {
+			stat, err := os.Stat(file)
+			if err == nil && stat.Mode()&(fs.ModeDevice|fs.ModeNamedPipe) != 0 {
```

Comment:
I don't know if we can/want to support heredoc as well.

Reply:
This should work, not entirely sure what the use case would be, but a heredoc becomes a pipe:

```
$ readlink -f /dev/stdin <<EOF
> test
> EOF
/proc/52258/fd/pipe:[232761]
```

Reply:
One may want to script it like :
```
promtool test rules alerts-test.yaml <<EOF
groups:
- name: myalerts
  rules:
  - alert: NoNewMessages
    expr: sum(increase(messages_sent_total[3m])) == 0
    annotations:
      summary: 'My component did not send any messages for 3 minutes'
EOF
```